Lib/socket.py

```diff
@@ -728,6 +728,78 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
     else:
         raise error("getaddrinfo returns an empty list")
 
+def bind_socket(address, family=AF_UNSPEC, type=SOCK_STREAM, *, backlog=128,
+                reuse_port=False, flags=getattr(_socket, "AI_PASSIVE", 0)):
+    """Convenience function which creates a socket bound to *address*
+    (a 2-tuple (host, port)) and return the socket object.
+
+    If *host* is an empty string or None all network interfaces are assumed.
+
+    If *family* is AF_UNSPEC or None the address family will be
+    determined from the *host* specified in *address*.
+
+    *type* should be either SOCK_STREAM or SOCK_DGRAM.
+
+    *backlog* is the queue size passed to socket.listen() and is ignored
+    for SOCK_DGRAM socket types.
+
+    *flags* is a bitmask for getaddrinfo().
+
+    >>> with bind_socket((None, 8000)) as server:
+    ...     while True:
+    ...         conn, addr = server.accept()
+    ...         # handle new connection
+    """
+    host, port = address
+    if host == "":
+        # https://mail.python.org/pipermail/python-ideas/2013-March/019937.html
+        host = None  # all interfaces
+    if not family:
+        family = AF_UNSPEC
+    reuse_addr = hasattr(_socket, 'SO_REUSEADDR') and \
+        os.name == 'posix' and sys.platform != 'cygwin'
+    if reuse_port and not hasattr(_socket, "SO_REUSEPORT"):
+        raise ValueError("SO_REUSEPORT not supported on this platform")
+    info = getaddrinfo(host, port, family, type, 0, flags)
+    if family == AF_UNSPEC:
+        # prefer AF_INET over AF_INET6
+        info.sort(key=lambda x: x[0] == AF_INET, reverse=True)
+
+    err = None
+    for res in info:
+        af, socktype, proto, canonname, sa = res
+        try:
+            sock = socket(af, socktype, proto)
+        except error as _:
+            err = _
```

Comment:
I understand why do you need this code but please rename `_` to something more meaningful.
Usually, I assume that `_` is for unused var which is not the case here.